src/vivarium_csu_ltbi/components/treatment.py

```diff
@@ -0,0 +1,121 @@
+                                                                    key_columns=['sex'],
+                                                                    value_columns=['value'])
+        return coverage_table_with_hiv, coverage_table_under_five_hhtb
+
+    def on_initialize_simulants(self, pop_data):
+        initialized = pd.DataFrame({'treatment_date': pd.NaT,
+                                    'treatment_type': 'untreated'}, index=pop_data.index)
+        self.population_view.update(initialized)
+
+    def on_time_step(self, event):
+        pop = self.population_view.get(event.index)
+
+        # Generate bit masks for our group conditions
+        with_hiv = self.get_hive_positive_subgroup(pop)
+        under_five_hhtb = self.get_under_five_hhtb_subgroup(pop)
+
+        # HIV+ Group
+        with_hiv_idx = pop.loc[with_hiv].index
+        with_hiv_threehp = self.three_hp_coverage_with_hiv(with_hiv_idx)
+        with_hiv_sixh = self.six_h_coverage_with_hiv(with_hiv_idx)
+        with_hiv_treatment_choices = self.treatment_stream.choice(with_hiv_idx, ['3HP', '6H', 'untreated'],
+                                                                  pd.concat([with_hiv_threehp, with_hiv_sixh,
+                                                                             (1 - with_hiv_threehp + with_hiv_sixh)],
+                                                                            axis=1))
+        with_hiv_treated = with_hiv_treatment_choices.loc[with_hiv_treatment_choices != 'untreated']
+        pop.loc[with_hiv_treated.index, 'treatment_date'] = event.time
+        pop.loc[with_hiv_treated.index, 'treatment_type'] = with_hiv_treated
+
+        # Under 5 and exposed to household TB group
+        under_five_hhtb_idx = pop.loc[under_five_hhtb].index
+        under_five_hhtb_threehp = self.three_hp_coverage_under_five_hhtb(under_five_hhtb_idx)
+        under_five_hhtb_sixh = self.six_h_coverage_under_five_hhtb(under_five_hhtb_idx)
+        under_five_hhtb_treatment_choices = self.treatment_stream.choice(under_five_hhtb_idx, ['3HP', '6H', 'untreated'],
+                                                                         pd.concat([under_five_hhtb_threehp,
+                                                                                    under_five_hhtb_sixh,
+                                                                                    (1 - under_five_hhtb_threehp + under_five_hhtb_sixh)],
+                                                                                   axis=1))
+        under_five_hhtb_treated = under_five_hhtb_treatment_choices.loc[under_five_hhtb_treatment_choices != 'untreated']
+        pop.loc[under_five_hhtb_treated.index, 'treatment_date'] = event.time
+        pop.loc[under_five_hhtb_treated.index, 'treatment_type'] = under_five_hhtb_treated
+
+        self.population_view.update(pop)
+
+    def get_hive_positive_subgroup(self, pop):
```

Comment:
Is there a reason for the 'e'? get_hivE_positive_subgroup versus get_hiv_positive_subgroup